src/cosalib/aws.py

```diff
@@ -111,6 +112,15 @@ def aws_run_ore_replicate(build, args):
 
 @retry(reraise=True, stop=stop_after_attempt(3))
 def aws_run_ore(build, args):
+    # set environment variables before setting up the ore command
+    image_yaml = flatten_image_yaml(
+        '/usr/lib/coreos-assembler/image-default.yaml',
+        flatten_image_yaml('src/config/image.yaml')
+    )
+    os.environ["aws_imds_support"] = image_yaml['aws-imds-support']
+    os.environ["aws_volume_type"] = image_yaml['aws-volume-type']
+    os.environ["aws_boot_mode"] = image_yaml['aws-boot-mode']
```

Comment:
Is there a reason we're not passing this info as parameters instead?

Reply:
Or, just pass the final `image.json` to mantle and have it parse it directly...

Reply:
yes. I view these settings as relatively temporary so my guidance to @prestist was to use environment variables rather than figure out how to weave in/pass the information down through each layer.

The goal here is to remove these settings after some relatively short period of time.

Maybe the advice was misguided. If so, it's on me.

Reply:
> Or, just pass the final `image.json` to mantle and have it parse it directly...

I thought about that, but it's less clear to me whether that's the best path forward. AFAIK, this would be a first for `ore`, which we've treated so far as a lower level tool wrapped by cosa. I think the value of keeping it separate is to have the `image.yaml` handling centralized at the cosa level, and keeping `ore` generic.

> Maybe the advice was misguided. If so, it's on me.

I'm not strongly against it, it just feels hacky. :) I think the "passing down through layers" wouldn't be hard since `CreateHVMImage()` is called directly by `upload.go` where the flags are located, so it'd just be one layer (and a cumbersome function signature, but like you said, it'd hopefully be temporary).

Reply:
> I'm not strongly against it, it just feels hacky. :) I think the "passing down through layers" wouldn't be hard since `CreateHVMImage()` is called directly by `upload.go` where the flags are located, so it'd just be one layer (and a cumbersome function signature, but like you said, it'd hopefully be temporary).

Right, but what's the goal? Is the goal to not use env variables at all here? In which case we'd need to add `ore` CLI args, which I'd rather avoid if we plan to delete it soon.

If we're going to use env vars anyway I think I'd rather it be where the PR currently has it than at the `upload.go` level so we don't have to change the function signatures.

TL;DR I definitely wouldn't suggest using env vars for this if it wasn't a temporary thing. I think in that case I'd advocate for `ore` CLI opts instead.

Reply:
What I'm suggesting is to not use env vars at all since it's trivial to do it with switches, which are cleaner and more characteristic of `ore`. Also, hacks we think are temporary today often turn out not to be and linger for years. :)

Reply:
> "switches" being CLI opts?
> 
> i.e. `ore aws --volumetype=gp2 ...` ?

Yup, exactly!

Reply:
Ok @jlebon I think I understand so it would be like this =? https://github.com/coreos/coreos-assembler/pull/3607